src/patterns/ec2-app/base.ts

```diff
@@ -276,6 +276,15 @@ export interface GuEc2AppProps extends AppIdentity {
      * @defaultValue /:STAGE/:stack/:app/google-auth-credentials
      */
     credentialsSecretsManagerPath?: string;
+
+    /**
+     * When using Auth in the ALB, which stage of cognito-lambda to use.
+     *
+     * For most applications this should always be PROD, even in the CODE environments.
+     *
+     * @defaultValue PROD
+     */
+    cognitoAuthStage?: "PROD" | "CODE";
```

Comment:
minor: In https://github.com/guardian/cdk/blob/main/docs/architecture-decision-records/006-stage-awareness.md the argument was made to not restrict `stage`, keeping it as a `string`. Does this union type contradict this? Granted the scope of this property is not the same as `stage` for an application. Might possible alternatives be to use a `string` type, or make this a `boolean`, e.g. `usePreProductionCognitoAuth`?